src/core/atomic.d

```diff
@@ -56,6 +56,70 @@ version( AsmX86 )
     {
         return addr % T.sizeof == 0;
     }
+
+    // Uses specialized asm for fast fetch and add operations
+    HeadUnshared!(T) atomicFetchAdd(T)( ref shared T val, T mod ) pure nothrow @nogc
+        if( __traits(isIntegral, T) )
+    in
+    {
+        // NOTE: 32 bit x86 systems support 8 byte CAS, which only requires
+        //       4 byte alignment, so use size_t as the align type here.
+        static if( T.sizeof > size_t.sizeof )
+            assert( atomicValueIsProperlyAligned!(size_t)( cast(size_t) &val ) );
+        else
+            assert( atomicValueIsProperlyAligned!(T)( cast(size_t) &val ) );
+    }
+    body
+    {
+        HeadUnshared!(T) oval = void;
+        static if (T.sizeof == 1)
+        {
+            asm pure nothrow @nogc
+            {
+                mov AL, mod;
+                mov RDX, val;
```

Comment:
Those `RDX` registers are for `X86_64` only, you need to copy&paste the code and use `EDX` for 32-bit `X86`.